### PR TITLE
Add .NET 8 targets

### PR DESCRIPTION
**Pull request by adamhathcock**
adjust csprojs
add nullability fixes

### CODE REVIEW

**Comment by adamhathcock** [comment]
Seeing this with ilpack and new targets though https://github.com/gluck/il-repack/issues/298

**Comment by adamhathcock** [comment]
il-repack isn't a problem on .NET 8

File: src/Speckle.Sdk/Credentials/AccountManager.cs [hunk 1]
```diff
@@ -606,7 +606,7 @@ private static async Task<string> GetAccessCode(string server, string challenge,
       throw new Exception("Local auth flow failed to complete within the timeout window");
     }
 
-    if (task.IsFaulted)
+    if (task.IsFaulted && task.Exception is not null)
```
Comment (AlanRynne) [inline src/Speckle.Sdk/Credentials/AccountManager.cs:609]: Is there a chance where the task `IsFaulted` but the `task.Exception` is null?

Would we not want to log that?

File: src/Speckle.Sdk/Api/GraphQL/Client.cs [hunk 2]
```diff
@@ -338,7 +338,7 @@ private static HttpClient CreateHttpClient(Account account)
     httpClient.DefaultRequestHeaders.Add("apollographql-client-name", Setup.ApplicationVersion);
     httpClient.DefaultRequestHeaders.Add(
       "apollographql-client-version",
-      Assembly.GetExecutingAssembly().GetName().Version.ToString()
+      Assembly.GetExecutingAssembly().GetName().Version?.ToString()
```
Comment (AlanRynne) [inline src/Speckle.Sdk/Api/GraphQL/Client.cs:341]: This is better than having a null ref exception but, do we ever expect this to be null?

**Review by AlanRynne — APPROVED** [review]
Some random questions, a bunch are not strictly related to your code changes.

File: src/Speckle.Sdk/Transports/ServerUtils/ProgressContent.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ public ProgressContent(HttpContent innerContent, Action<ProgressArgs>? progress)
     innerContent.Headers.CopyTo(Headers);
   }
 
-  protected override Task SerializeToStreamAsync(Stream stream, TransportContext context)
+  protected override Task SerializeToStreamAsync(Stream stream, TransportContext? context)
   {
     ProgressStream progressStream = new(stream, _innerContent.Headers.ContentLength, _progress, false);
     return _innerContent.CopyToAsync(progressStream);
```
[the remark this answers is not in the document]
Comment (JR-Morgan) [reply src/Speckle.Sdk/Transports/ServerUtils/ProgressContent.cs:21]: We're not using, but its needed for the override.

**Comment by JR-Morgan** [comment]
Please can we get https://github.com/specklesystems/speckle-sharp-sdk/pull/92 in before this one...

Unfortunately tho, once #92 is merged, there will be lots of new net8 analysers now running, which will complain on PR that we'll need to tweak and supress a few.

File: src/Speckle.Sdk/Serialisation/SpeckleObjectDeserializer.cs [hunk 4]
```diff
@@ -134,7 +134,7 @@ public Task<Base> Deserialize(string rootObjectJson)
       {
         case JsonToken.PropertyName:
           {
-            var propName = reader.Value.NotNull().ToString();
+            string propName = (reader.Value?.ToString()).NotNull();
```
[the remark this answers is not in the document]
Comment (adamhathcock) [reply src/Speckle.Sdk/Serialisation/SpeckleObjectDeserializer.cs:137]: ToString() returns nullables now and I didn't want to do NotNull twice

File: src/Speckle.Sdk/Credentials/AccountManager.cs (hunk 1 above)
Comment (adamhathcock) [reply src/Speckle.Sdk/Credentials/AccountManager.cs:607]: I don't think there's a practical chance of it.  An Exception is always thrown

**Review by adamhathcock — COMMENTED** [review]
This is ready now

**Review by AlanRynne — APPROVED** [review]

**Pull request merged**